packages/tx/src/baseTransaction.ts

```diff
@@ -90,23 +91,41 @@ export abstract class BaseTransaction<TransactionObject> {
     const vB = toBuffer(v === '' ? '0x' : v)
     const rB = toBuffer(r === '' ? '0x' : r)
     const sB = toBuffer(s === '' ? '0x' : s)
-
     const nonceB = toBuffer(nonce === '' ? '0x' : nonce)
-    if (nonceB.length > 8) {
-      // Implements EIP-2681 which limits nonce to 2^64-1
-      // https://eips.ethereum.org/EIPS/eip-2681
-      throw new Error('nonce cannot be more than 2^64-1')
+    const gasLimitB = toBuffer(gasLimit === '' ? '0x' : gasLimit)
+    const valueB = toBuffer(value === '' ? '0x' : value)
+
+    const integerProps = {
+      nonce: nonceB,
+      gasLimit: gasLimitB,
+      value: valueB,
+      v: vB,
+      r: rB,
+      s: sB,
```

Comment:
I guess this is not working here in the constructor, since this "no leading zeros" rule really only applies for Txs which have been RLP encoded before. This is only be the case for Txs initialized with the `fromSerializedTx()` (and, alias: `fromRLPSerializedTx()`) and then the `fromValuesArray()` constructors (for the latter one I am only 95% sure that this is the case, might be good if someone confirms. If so we should take this function as a point for adding the checks since all others call into `fromValuesArray()`.

Let's take the situation here and pick out e.g. the `nonce`. As a final value nonce can very well be `0` (actually that's the base/first possible value), same for `v` values for non-legacy txs.

The `nonce` parameter is of type `BNLike` and can be e.g. a `number` and a `Buffer`.

An input value of `nonce = 0` is valid here, but also an empty Buffer. Both translate to a BN value of `0` at the end, see the following for illustration:

```shell
> import { BN, toBuffer } from 'ethereumjs-util'
{}
> let nonceN = 0
undefined
> let nonceB = Buffer.from([])
undefined
> toBuffer(nonceN)
<Buffer 00>
> new BN(toBuffer(nonceN))
<BN: 0>
> toBuffer(nonceB)
<Buffer >
> new BN(toBuffer(nonceN))
<BN: 0>
```

So these different `nonce` inputs lead to different intermediary Buffer values (`<Buffer 00>` and `<Buffer >`). If we are doing the checks from above here, we are unrightfully disallowing the case with the empty Buffer.

So we need to really dedicatedly go into the different `fromValuesArray()` constructors. Maybe we can add a helper function in `util.ts` on this to not do too much code duplication.

Reply:
Indeed, I wish I had read this last night :grimacing: I independently came to this same realization this morning after much exploring rabbit trails.